lighthouse-core/gather/driver.js

```diff
@@ -743,6 +743,20 @@ class Driver {
   }
 
   /**
+   * Returns the flattened list of all DOM nodes within the document.
+   * @param {boolean=} pierce Whether to pierce through shadow trees and iframes.
+   *     True by default.
+   * @return {!Promise<!Array<!Element>>} The found elements, or [], resolved in a promise
+   */
+  getAllElementsInDocumentAndFrames(pierce = true) {
```

Comment:
In that case, I think we should go back to what I had originally: `getAllElementsInDocument(pierce = true)`. 

Closest to the DT protocol for both func and param naming. The docs already explain what `pierce` does ("pierces through iframes and shadow roots when true"). It doesn't need to be in the method name.